src/views/open-api-docs-view/components/open-api-sidebar-contents/index.tsx

```diff
@@ -23,64 +24,28 @@ export function OpenApiSidebarContents({
 }: {
 	navItems: OpenApiNavItem[]
 }) {
-	const { asPath } = useRouter()
+	const [filterValue, setFilterValue] = useState('')
 
-	/**
-	 * Note next/router `asPath` returns *with* the `#hash`.
-	 *
-	 * Here, we remove the hash, so that we can compare to the `fullPath`
-	 * of each nav item, which are not expected to have hashes,
-	 * to the URL path that we're on.
-	 *
-	 * Note that domain does not matter, as we're just grabbing the full path.
-	 */
-	const urlPathname = useMemo(() => newUrl(asPath).pathname, [asPath])
+	// Highlight active navItems, including `#hash` links via `useActiveSection`.
+	const navItemsWithActive = useNavItemsWithActive(navItems)
 
-	/**
-	 * Build an array of section slugs,
-	 * and pass these to `useActiveSection`.
-	 */
-	const sectionSlugs = useMemo(() => {
```

Comment:
I've split all the "active section" and highlighting logic out to a separate file for clarity.

"Split" view is probably more useful for this file 👍